README
Name
    App::Munchies - Catalyst example application using food recipes as a
    data set

Version
    0.7.$Revision: 1318 $

Synopsis
       # Start the development server with

       bin/munchies_server -d -r -rd 1 -rr "\\.xml\$|\\.pm\$" \
          --restart_directory lib

       # Start the production server with

       plackup -s Starman --access-log var/logs/starman_server.log \
          bin/munchies_psgi

Description
    This is an example application for the CatalystX::Usul base class

    Some web applications require common controllers and data models. For
    example; welcome mat, authentication, password changing, navigation
    tools and site map are some of the controllers implemented here. Since
    these modules are not an end in themselves, most of the visible text is
    stored in XML configuration files and can be customised using the
    controllers and data models provided. The string "Munchies" for example
    is a generic application name and it is meant to be replaced with a more
    specific application name once it has been written

    Pages are rendered using a single TT template. The template, CSS,
    Javascript and collection of GIFs, PNG, etc are stored together and form
    a "skin" for which a switching mechanism is provided. This enables
    development of the web application to proceed independently of the
    interface development. Content (provided by Perl on Catalyst) has been
    separated from layout (TT+CSS), presentation (CSS) and behaviour (JS).
    If the interface is not to your liking, write your own skin. These ideas
    and techniques have been aquired from <http://www.csszengarden.com/> and
    <http://www.cssplay.co.uk/>

    Don't even think about using anything other than a modern version of
    Chrome / Firefox / Opera to display these pages

Dependencies
    Either Apache/mod_perl or Plack/Starman is required to serve HTTP (the
    other Catalyst engines are also supported). Requires either PostgreSQL
    or MySQL to be installed. The XML parser requires "libxml2" and
    "libxml2-dev" otherwise a slow pure Perl implementation will be used.
    Installing Perl module dependencies from CPAN will require "make", "gcc"
    and "g++" (or equivalents) to be installed. GD::SecurityImage (used to
    generate Captchas) depends on "libgd2-noxpm" and "libgd2-noxpm-dev". It
    also requires the "StayPuft.ttf" font to be installed

Installation
    Run these commands as root to install this application from a
    distribution tarball:

       tar -xvzf App-Munchies-?.?.?.tar.gz
       cd App-Munchies-?.?.?
       ./install.sh

    It defaults to installing all files (including the var data) under
    /opt/app-munchies (which is easy to remove if this is not a permanent
    installation)

    Once the schema has been deployed and populated the following (optional)
    commands will be run:

       bin/munchies_cli    -nc pod2html
       bin/munchies_schema -nc catalog_mmf

    as the *munchies* user. They may take some time to finish. When complete
    the var area of the application is about 60Mb in size

    This distribution contains a setuid root program. It is used to provide
    limited access to root only functions, e.g. authentication against
    /etc/shadow. The build process asks if this should be enabled. It is not
    enabled by default

    N.B. Remove *user_root* from var/secure/support.sub if it exists

    N.B. Change the password for the admin account in the R00-Internal realm

Configuration and Environment
    Application configuration is in the file
    lib/App/Munchies/app_munchies.xml

Diagnostics
    The "-d" option on the bin/munchies_server.pl command line starts the
    development server server in debug mode

Subroutines/Methods
    None

Incompatibilities
    Cygwin - Has a wierd gecos field in the passwd file that is a problem
    for the identity model.

    The Pod::ProjectDocs module will not install without forcing.
    CatalystX::Usul::ProjectDocs monkey patches Pod::ProjectDocs with a
    different syntax highlighter so that the post install commands can
    generate the HTML version the application documentation

Bugs and Limitations
    There are no known bugs in this module. Please report problems to the
    address below. Patches are welcome

Author
    Peter Flanigan, "<Support at RoxSoft.co.uk>"

Acknowledgements
    Larry Wall - For the Perl programming language

License and Copyright
    Copyright (c) 2012 Peter Flanigan. All rights reserved

    This program is free software; you can redistribute it and/or modify it
    under the same terms as Perl itself. See perlartistic

    This program is distributed in the hope that it will be useful, but
    WITHOUT WARRANTY; without even the implied warranty of MERCHANTABILITY
    or FITNESS FOR A PARTICULAR PURPOSE



Changes
Revision history for App::Munchies

{{ $NEXT }}

v0.7.1318 2012-04-22 18:12:12 BST
          - Turned CPAN testing on
          - Updated for H::FW::Cloud api change
          - Bumped CX::U dependency to version 0.7
          - Updated inc::Bob to 1.3
          - Fixed homepage link in Build.PL

v0.6.1307 2012-04-15 19:33:35 BST
          - Improved Bingos broken toolchain regex to include more smokers

v0.6.1305 2012-04-04 19:32:27 BST
          - Bingos smoker is still failing to install Sort::Naturally. Added
            to broken toolchain function

v0.6.1302 2012-04-04 13:07:47 BST
          - Turning CPAN testing off because
            08289c92-7e3f-11e1-9d6f-f6dbfa7543f5. Thats 24hrs after peregrins
            Crixa upload and smokers are still failing because cpanmetadb has
            Sort::Naturally as not found
          - Bumped version numbers of files in the bin directory

v0.6.1301 2012-04-04 09:11:26 BST
          - inc::Bob now requires CX::U::Build after Perl version test
          - Updated inc::Bob to 1.2

v0.5.1285 2012-03-28 12:58:42 BST
          - Updated inc::Bob. Added inc::CPANTesting
          - Updated for Catalyst 5.9
          - Updated Moose / Class::MOP / MRO::Compat dependencies
          - Added message translation method to CLI
          - Moved boilerplate M::B subclass code to inc/Bob.pm
          - Suppressed Konig CPAN testing again
          - Added extended toolchain
          - Switched to Catalyst 5.8. Uses Moose

v0.4.808  2009-07-14
          - Uses CX::Usul::Constants
          - Build.PL no longer uses the M::B::notes attribute
          - Now uses C::P::ContextualUriFor
          - Fixed library/admin uri warning
          - Fixed users/admin uri warning
          - Only uses CX::Usul::uri_for
          - Fixed bad DDL file names again

v0.4.794  30/06/2009
          - Build.PL no longer needs APPLDIR

v0.3.786  29/06/2009
          - Renamed suid to admin and misc to cli
          - Adapted to Catalyst 5.8
          - Fixed no index directory value type in META.yaml

v0.2.750  09/06/2009
          - Worked around bug in DBIx::Class::Schema::deploy
          - Improved testing idioms
          - Now sets resources attribute in META.yml
          - Paragraph columns now have a max width


Build.PL
# @(#)$Id: Build.PL 1318 2012-04-22 17:10:47Z pjf $

use strict;
use warnings;
use inc::Bob; # Module::Build subclass
use version; our $VERSION = qv( sprintf '0.7.%d', q$Rev: 1318 $ =~ /\d+/gmx );

use File::Spec::Functions;

Bob->new( {
   module     => q(App::Munchies),
   home_page  => q(http://www.roxsoft.co.uk:8080/library/catalog),
   requires   => {
      'perl'                                 => 5.010,
      'namespace::autoclean'                 => 0.09,
      'parent'                               => 0.221,
      'version'                              => 0.87,
      'Catalyst'                             => 5.90010,
      'CatalystX::RoleApplicator'            => 0.005,
      'CatalystX::Usul'                      => q(0.7.1184),
      'Class::Accessor'                      => 0.31,
      'Class::Null'                          => 1.09,
      'DBIx::Class'                          => 0.08112,
      'Data::CloudWeights'                   => q(0.7.0),
      'Date::Discordian'                     => 1.36,
      'DateTime::Event::Sunrise'             => 0.0501,
      'DateTime::Fiction::JRRTolkien::Shire' => 0.02,
      'File::DataClass'                      => q(0.9.0),
      'HTML::Accessors'                      => q(0.7.0),
      'HTML::Entities'                       => 3.68,
      'KinoSearch'                           => 0.3,
      'Moose'                                => 2.0402,
      'Plack'                                => 0.9985,
      'Starman'                              => 0.3000,
      'Text::Lorem::More'                    => 0.13,
      'TryCatch'                             => 1.002000,
      'WWW::Wikipedia'                       => 2.00, },
   recommends                                => {
      'local::lib'                           => 1.008004,
      'Plack::Middleware::Debug'             => 0.11,
      'SVN::Class'                           => 0.16, },
   configure_requires        => {
      'version'              => 0.87,
      'CatalystX::Usul'      => q(0.6.0),
      'Module::Build'        => 0.38, },
   build_requires            => {
      'version'              => 0.87,
      'CatalystX::Usul'      => q(0.6.0),
      'Test::More'           => 0.74, },
   no_index_dir              => [ qw(examples inc t), catdir( qw(var root) ) ],
   notes                     => {
      'admin_role'           => q(support),
      'copy_files'           => [ {
         from                => [ qw(META.yml)         ],
         to                  => [ qw(var etc META.yml) ] }, {
         from                => [ qw(var etc etc_default) ],
         to                  => [ q(), qw(etc default app-munchies) ] }, ],
      'create_dirs'          => [ [ q(), qw(etc default) ],
                                  [ qw(var hist) ],
                                  [ qw(var logs) ],
                                  [ qw(var root recipes) ],
                                  [ qw(var root reports) ],
                                  [ qw(var run) ],
                                  [ qw(var tmp) ], ],
      'create_files'         => [ [ qw(var etc  aliases) ],
                                  [ qw(var logs cli.log) ],
                                  [ qw(var logs admin.log) ],
                                  [ qw(var logs server.log) ], ],
      'database_name'        => q(library),
      'debian_depends'       => [ qw(libexpat1-dev perl-suid) ],
      'deploy_server'        => q(plack),
      'group'                => q(munchies),
      'll_author'            => q(A/AP/APEIRON),
      'll_ver_dir'           => q(local-lib-1.008004),
      'make_default'         => 1,
      'min_perl_ver'         => q(5.008),
      'owner'                => q(munchies),
      'perl_mirror'          => q(http://ftp.plig.net/pub/CPAN/),
      'post_install_cmd'     => q(munchies_admin -nc post_install),
      'post_install_methods' => [ { admin  => q(create_ugrps)    },
                                  { admin  => q(set_owner)       },
                                  { admin  => q(set_permissions) },
                                  { admin  => q(init_suid)       },
                                  { schema => q(create_schema)   },
                                  { admin  => q(make_default)    },
                                  { admin  => q(deploy_server)   },
                                  { admin  => q(restart_server)  },
                                  { cli    => q(pod2html)        },
                                  { schema => q(catalog_mmf)     }, ],
      'prefix'               => q(munchies),
      'process_owner'        => q(munchies),
      'restart_server'       => 1,
      'restart_server_cmd'   => q(/usr/sbin/invoke-rc.d app-munchies restart),
      'shell'                => q(/bin/ksh),
      'uninstall_cmd'        => q(munchies_admin -nc uninstall),
   },
} )->create_build_script; # Goal!

# Local Variables:
# mode: perl
# tab-width: 3
# End:
